docs/diagrams/DoneClassDiagram.puml

```diff
@@ -0,0 +1,71 @@
+@startuml
+class DoneCommandParser implements Parser {
+    /' Methods '/
+    + {static} parse(args : String) : DoneCommand
+}
+
+interface Parser<T> {
+    parse(userInput : String) : DoneCommand
+}
+
+abstract class Command {
+    {abstract} execute(Model model) : CommandResult
+}
+
+class DoneCommand extends Command {
+    /' Fields '/
+    + {static} COMMAND_WORD : String
+    + {static} MESSAGE_USAGE : String
+    + {static} MESSAGE_DELIVERED_SUCCESS : String
+    + {static} MESSAGE_ORDER_ALREADY_DELIVERED : String
+    - targetIndex : Index
+    - doneOrderDescriptor : DoneCommand.DoneOrderDescriptor
+
+    /' Methods '/
+    + execute(model : Model) : CommandResult
+    + equals(Object other) : boolean
+    - {static} createEditedOrder(Order orderToDeliver, DoneOrderDescriptor doneOrderDescriptor) : Order
+}
+
+DoneCommand -down-* DoneOrderDescriptor
+
+class DoneOrderDescriptor {
+    /' Fields '/
+    - tid : TransactionId;
+    - name : Name;
+    - phone : Phone;
+    - email : Email;
+    - address : Address;
+    - timeStamp : TimeStamp;
+    - warehouse : Warehouse;
+    - cod : CashOnDelivery;
+    - comment : Comment;
+    - itemType : TypeOfItem;
+    - deliveryStatus : boolean;
```

Comment:
Might want to ensure that the output diagram gives the right symbols (-, +) for the private and public fields.